insights/specs/datasources/containers_inspect.py

```diff
@@ -0,0 +1,94 @@
+"""
+Custom datasources for containers inspect information
+"""
+from insights.core.context import HostContext
+from insights.core.dr import SkipComponent
+from insights.core.plugins import datasource
+from insights.core.spec_factory import DatasourceProvider, foreach_execute
+from insights.core.filters import get_filters
+from insights.specs import Specs
+import json
+from insights.specs.datasources.container import running_rhel_containers
+
+
+@datasource(running_rhel_containers, HostContext)
+def running_rhel_containers_id(broker):
+    """
+    Returns a list of container_id of the running containers.
+    """
+    containers_info = []
+    for container in broker[running_rhel_containers]:
+        containers_info.append((container[1], container[2]))
+    if containers_info:
+        return containers_info
+    raise SkipComponent
+
+
+class LocalSpecs(Specs):
+    """ Local specs used only by docker|podman_inspect datasources """
+
+    containers_inspect_data_raw = foreach_execute(running_rhel_containers_id, "/usr/bin/%s inspect %s")
+    """ Returns the output of command ``/usr/bin/docker|podman inspect <container ID>`` """
+
+
+@datasource(LocalSpecs.containers_inspect_data_raw, HostContext)
+def containers_inspect_data_datasource(broker):
+    """
+    This datasource provides the filtered information collected
+    from ``/usr/bin/docker|podman inspect <container ID>``.
+
+    Typical content of ``/usr/bin/docker|podman inspect <container ID>`` file is::
+
+        [
+            {
+                "Id": "aeaea3ead52724bb525bb2b5c619d67836250756920f0cb9884431ba53b476d8",
+                "Created": "2022-10-21T23:47:24.506159696-04:00",
+                "Path": "sleep"
+                ...
+            }
+            ...
+        ]
+
+    Returns:
+        list: item is JSON string containing filtered information.
+
+    Raises:
+        SkipComponent: When the filter/path does not exist or any exception occurs.
+    """
+
+    def _find(d, tag, path):
+        if tag in d:
+            yield d[tag]
```

Comment:
I don't think it's correct to make this `_find` a generator, since it may miss the keys which is at the end of the `filters` but locate in the top of the `json`.